gp/cartesian_graph.py

```diff
@@ -1,53 +1,53 @@
 import collections
+import gp
 import sympy
 import torch  # noqa: F401
 
 from .node import InputNode, OutputNode
 from .exceptions import InvalidSympyExpression
+from typing import Callable, List, Optional
 
 
 class CartesianGraph:
     """Class representing a particular Cartesian graph defined by a
     Genome.
     """
-
-    def __init__(self, genome):
+    def __init__(self, genome: gp.Genome) -> None:
         """Init function.
 
         Parameters
         ----------
         genome: Genome
             Genome defining graph connectivity and node operations.
         """
-        self._n_outputs = None
-        self._n_inputs = None
-        self._n_columns = None
-        self._n_rows = None
-        self._nodes = None
-        self._gnome = None
+        self._n_outputs: int = 0
```

Comment:
is there a `nan` type for ints that could signal uninitialized values?

Reply:
I now changed this and don't initalize the values at all.

Reply:
my C++ alarms are ringing reading this statement ;) how does it work in python? to which value are they set in that case? like in C++ whatever resides at that memory location? defaults of the annotations module?

Reply:
Yeah, maybe that is not a good thing, Python will throw a `NameError` when you try to access the member. The alternative would be to write it like this:
`self._n_outputs: Optional[int] = None` (i.e. to make the type Optional which means it can be None or an integer).